File: app/src/main/java/au/com/raicovtechnologyservices/musaique/Song.java
package au.com.raicovtechnologyservices.musaique;

import android.content.ContentUris;
import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.media.Image;
import android.net.Uri;
import android.os.ParcelFileDescriptor;

import java.io.FileDescriptor;

/**
 * Created by Jamie on 10/04/2017.
 */

public class Song {

    private String trackTitle;
    private String artistName;
    private String albumName;
    private String duration;
    private String songPath;
    private long id;
    private long albumId;
    private Context context;



    private int position;

    public Song(long id, String trackTitle, String artistName, String albumName, long albumId, String duration, String songPath, Context context, int position){
        this.id = id;
        this.trackTitle = trackTitle;
        this.artistName = artistName;
        this.albumName = albumName;
        this.albumId = albumId;
        this.duration = duration;
        this.context = context;
        this.songPath = songPath;
        this.position = position;
        getAlbumArt();
    }

    public String getTrackTitle() {
        return trackTitle;
    }

    public String getArtistName() {
        return artistName;
    }

    public String getSongPath() { return songPath; }

    public String getAlbumName() {
        return albumName;
    }

    public String getDuration() {
        return duration;
    }

    public Bitmap getAlbumArt()
    {
        Bitmap bm = null;
        try
        {
            final Uri sArtworkUri = Uri
                    .parse("content://media/external/audio/albumart");

            Uri uri = ContentUris.withAppendedId(sArtworkUri, this.albumId);

            ParcelFileDescriptor pfd = context.getContentResolver()
                    .openFileDescriptor(uri, "r");

            if (pfd != null)
            {
                FileDescriptor fd = pfd.getFileDescriptor();
                bm = BitmapFactory.decodeFileDescriptor(fd);
            }
        } catch (Exception e) {
        }
        return bm;
    }

    public int getPosition() {
        return position;
    }
}

File: app/src/main/java/au/com/raicovtechnologyservices/musaique/Song.kt
package au.com.raicovtechnologyservices.musaique

import android.content.ContentUris
import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.media.Image
import android.net.Uri
import android.os.ParcelFileDescriptor

import java.io.FileDescriptor


class Song(private val id: Long, val trackTitle: String, val artistName: String, val albumTitle: String, private val albumId: Long, val songPath: String, private val context: Context, private val position: Int) {

    init {
        albumArt
    }


    val albumArt: Bitmap?
        get() {
            var bm: Bitmap? = null
            try {
                val sArtworkUri = Uri
                        .parse("content://media/external/audio/albumart")

                val uri = ContentUris.withAppendedId(sArtworkUri, this.albumId)

                val pfd = context.contentResolver
                        .openFileDescriptor(uri, "r")

                if (pfd != null) {
                    val fd = pfd.fileDescriptor
                    bm = BitmapFactory.decodeFileDescriptor(fd)
                }
            } catch (e: Exception) {
            }

            return bm
        }

}
